scipy/stats/_continuous_distns.py

```diff
@@ -6373,7 +6373,8 @@ def _isf(self, q, c):
         return np.where(q > 0.5, (2.0*(1.0 - q))**(1.0/c), (2*q)**(-1.0/c))
 
     def _munp(self, n, c):
-        return c**2 / (c**2 - n**2)
+        with np.errstate(divide='ignore'):
+            return np.where(np.abs(n) < c, c**2 / (c**2 - n**2), np.inf)
```

Comment:
The doc of the `moment` function requires `n` to be a positive integer, and raises a `ValueError` if it’s not.

The above implementation is more general and correctly returns $\mathbb{E}[X^n]$ for any real number `n`. The advantage is that it doesn’t rely on the (implicit) pre-check of argument passed, and readily works should the argument restriction on `moment` function be relaxed.

Reply:
I know, this was just a polite nudge to simplify since we're not relying on or testing the more general behavior. As a compromise, I will adjust it to compare `c**2` and `n**2`, which have already been calculated.